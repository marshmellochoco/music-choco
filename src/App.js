// dependancy import
import { useState } from "react";
import { Route, BrowserRouter as Router, Switch } from "react-router-dom";

// component import
import { Player } from "./Components/AudioPlayer/Player";
import { Queue } from "./Pages/Queue/Queue";
import { Album } from "./Pages/Album/Album";
import { Home } from "./Pages/Home/Home";
import { EditAlbum } from "./Pages/AddSong/EditAlbum";
import { Login } from "./Pages/Login/Login";
import { useSelector } from "react-redux";

function App() {
    const [randomQueue, setRandomQueue] = useState([]);
    const authToken = useSelector((state) => state.authReducer.token);

    return (
        <div className="App" onContextMenu={(e) => e.preventDefault()}>
            {authToken !== "null" && authToken ? (
                <div className="app-container">
                    <Router basename="/">
                        <div className="page-container">
                            <Switch>
                                <Route exact path="/">
                                    <Home />
                                </Route>
                                <Route path="/albums/:id">
                                    <Album />
                                </Route>
                                <Route path="/add">
                                    <EditAlbum />
                                </Route>
                            </Switch>
                        </div>
                    </Router>
                    <Queue setRandomQueue={setRandomQueue} />
                    <Player randomQueue={randomQueue} />
                </div>
            ) : (
                <Login />
            )}
        </div>
    );
}

export default App;

// TODO: Song already in queue
// TODO: Homepage can add recommendation
// TODO: Pull song from album to queue or playlists

// TODO: Login
// TODO: Add song list, favorite
