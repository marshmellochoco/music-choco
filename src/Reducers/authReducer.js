const initState = {
    token: localStorage.getItem("TOKEN"),
};

export const authReducer = (state = initState, action) => {
    switch (action.type) {
        case "SET_TOKEN":
            localStorage.setItem("TOKEN", state.token);
            return {
                token: action.token,
            };
        case "RESET_TOKEN":
            localStorage.setItem("TOKEN", null);
            return {
                token: "",
            };
        default:
            return state;
    }
};
