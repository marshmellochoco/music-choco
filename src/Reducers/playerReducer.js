const initState = {
    playing: false,
    loop: localStorage.getItem("playerLoop") === "true",
    random: localStorage.getItem("playerRandom") === "true",
    volume:
        localStorage.getItem("playerVolume") &&
        localStorage.getItem("playerVolume") !== "null"
            ? parseFloat(localStorage.getItem("playerVolume"))
            : 1,
};

export const playerReducer = (state = initState, action) => {
    switch (action.type) {
        case "SET_PLAYING":
            return {
                ...state,
                playing: action.playing,
            };
        case "SET_VOLUME":
            localStorage.setItem("playerVolume", action.volume);
            return {
                ...state,
                volume: action.volume,
            };
        case "TOGGLE_LOOP":
            localStorage.setItem("playerLoop", !state.loop);
            return {
                ...state,
                loop: !state.loop,
            };
        case "TOGGLE_RANDOM":
            localStorage.setItem("playerRandom", !state.random);
            return {
                ...state,
                random: !state.random,
            };
        default:
            return state;
    }
};
