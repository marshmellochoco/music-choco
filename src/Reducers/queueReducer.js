const initState = {
    queue: [],
    queueData: [],
    loading: false,
};

export const queueReducer = (state = initState, action) => {
    switch (action.type) {
        case "SET_QUEUE":
            return {
                ...state,
                queue: [...action.queue],
            };
        case "ADD_QUEUE":
            if (state.queue.includes(action.songId)) {
                return state;
            }
            return {
                ...state,
                queue: [...state.queue, action.songId],
            };
        case "SET_LOADING":
            return {
                ...state,
                loading: action.loading,
            };
        case "SET_QUEUE_DATA":
            return {
                ...state,
                queueData: action.queueData,
            };
        case "RESET_QUEUE":
            return {
                initState,
            };
        default:
            return state;
    }
};
