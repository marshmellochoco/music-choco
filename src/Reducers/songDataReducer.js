const initState = {
    songData: {
        songId: "",
        title: "",
        duration: 0,
        albumname: "",
        albumId: "",
        artist: "",
    },
};

export const songDataReducer = (state = initState, action) => {
    switch (action.type) {
        case "SET_SONG_DATA":
            if (!action.songData) return state;
            return {
                ...state,
                songData: action.songData,
            };
        case "SET_PLAYING_SONG":
            return {
                ...state,
                songData: { ...state.songData, songId: action.songId },
            };
        case "RESET_SONG_DATA":
            return {
                initState,
            };
        default:
            return state;
    }
};
