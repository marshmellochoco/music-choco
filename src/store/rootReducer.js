import { combineReducers } from "redux";
import queueReducer from "./queue/queueReducer";
import playerReducer from "./player/playerReducer";
import userReducer from "./user/userReducer";

const rootReducer = combineReducers({
    playerReducer,
    queueReducer,
    userReducer,
});

export default rootReducer;
