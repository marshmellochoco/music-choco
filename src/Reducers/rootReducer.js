import { queueReducer } from "../Reducers/queueReducer";
import { songDataReducer } from "../Reducers/songDataReducer";
import { playerReducer } from "../Reducers/playerReducer";
import { authReducer } from "../Reducers/authReducer";

import { combineReducers } from "redux";

export const rootReducer = combineReducers({
    queueReducer,
    songDataReducer,
    playerReducer,
    authReducer,
});
