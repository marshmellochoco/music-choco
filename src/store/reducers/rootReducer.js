import { queueReducer } from "./queueReducer";
import { songDataReducer } from "./songDataReducer";
import { playerReducer } from "./playerReducer";
import { authReducer } from "./authReducer";

import { combineReducers } from "redux";

export const rootReducer = combineReducers({
    queueReducer,
    songDataReducer,
    playerReducer,
    authReducer,
});
