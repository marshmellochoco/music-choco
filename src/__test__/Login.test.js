import { createStore } from "redux";
import { Provider } from "react-redux";
import App from "../App";
import renderer from "react-test-renderer";
import { rootReducer } from "../Reducers/rootReducer";

const store = createStore(rootReducer);
store.dispatch({ type: "SET_TOKEN", token: "someToken" });

describe("Login", () => {
    it("should log in and redirect to homepage", () => {
        const tree = renderer.create(
            <Provider store={store}>
                {<App/>}
            </Provider>
        ).toJSON();
        expect(tree).toMatchSnapshot();
    });
});
