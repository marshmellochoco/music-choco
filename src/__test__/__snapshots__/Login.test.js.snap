// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login should log in and redirect to homepage 1`] = `
<div
  className="App"
  onContextMenu={[Function]}
>
  <div
    className="app-container"
  >
    <div
      className="page-container"
    >
      <div>
        <div
          className="headers"
        >
          <h1>
            music-choco
          </h1>
          <div
            className="search"
          >
            <svg
              className="icon"
              viewBox="0 0 24 24"
            >
              <path
                d="M9.5,3A6.5,6.5 0 0,1 16,9.5C16,11.11 15.41,12.59 14.44,13.73L14.71,14H15.5L20.5,19L19,20.5L14,15.5V14.71L13.73,14.44C12.59,15.41 11.11,16 9.5,16A6.5,6.5 0 0,1 3,9.5A6.5,6.5 0 0,1 9.5,3M9.5,5C7,5 5,7 5,9.5C5,12 7,14 9.5,14C12,14 14,12 14,9.5C14,7 12,5 9.5,5Z"
              />
            </svg>
            <input
              onChange={[Function]}
              type="text"
            />
          </div>
        </div>
        <div
          className="content"
        >
          Hmm... what to put here
          <br />
          <a
            href="/add"
            onClick={[Function]}
          >
            Add some song!
          </a>
          <br />
          <a
            href="/"
            onClick={[Function]}
          >
            Logout
          </a>
        </div>
      </div>
    </div>
    <div
      className="queue"
    >
      <div
        className="queueHeader"
      >
        <h1>
          Queue
        </h1>
        <div
          className="controls"
        >
          <svg
            className="icons "
            height="24px"
            onClick={[Function]}
            viewBox="0 0 24 24"
            width="24px"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M10.59 9.17L5.41 4 4 5.41l5.17 5.17 1.42-1.41zM14.5 4l2.04 2.04L4 18.59 5.41 20 17.96 7.46 20 9.5V4h-5.5zm.33 9.41l-1.41 1.41 3.13 3.13L14.5 20H20v-5.5l-2.04 2.04-3.13-3.13z"
            />
          </svg>
          <svg
            className="icons "
            height="24px"
            onClick={[Function]}
            viewBox="0 0 24 24"
            width="24px"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M12 4V1L8 5l4 4V6c3.31 0 6 2.69 6 6 0 1.01-.25 1.97-.7 2.8l1.46 1.46C19.54 15.03 20 13.57 20 12c0-4.42-3.58-8-8-8zm0 14c-3.31 0-6-2.69-6-6 0-1.01.25-1.97.7-2.8L5.24 7.74C4.46 8.97 4 10.43 4 12c0 4.42 3.58 8 8 8v3l4-4-4-4v3z"
            />
          </svg>
        </div>
      </div>
      <ul
        className="queueList"
        data-rbd-droppable-context-id="0"
        data-rbd-droppable-id="droppable"
      />
    </div>
    <div
      className="player"
      tabIndex="0"
    >
      <div>
        <div
          style={
            Object {
              "height": "0",
              "width": "0",
            }
          }
        />
        <div
          className="player-container"
        >
          <div
            className="player-playing"
          >
            <div
              className="icon"
            >
              <img
                alt=""
                onError={[Function]}
                src=""
              />
            </div>
            <div
              className="title"
            >
              ---
            </div>
            <div
              className="artist"
            >
              ---
            </div>
          </div>
          <div
            className="player-control"
          >
            <svg
              className="control-button prev"
              onClick={[Function]}
              viewBox="0 0 24 24"
            >
              <path
                d="M6,18V6H8V18H6M9.5,12L18,6V18L9.5,12Z"
                fill="currentColor"
              />
            </svg>
            <svg
              className="control-button play"
              fill="none"
              onClick={[Function]}
              stroke="currentColor"
              viewBox="0 0 24 24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M14.752 11.168l-3.197-2.132A1 1 0 0010 9.87v4.263a1 1 0 001.555.832l3.197-2.132a1 1 0 000-1.664z"
                strokeLinecap="round"
                strokeLinejoin="round"
                strokeWidth={2}
              />
              <path
                d="M21 12a9 9 0 11-18 0 9 9 0 0118 0z"
                strokeLinecap="round"
                strokeLinejoin="round"
                strokeWidth={2}
              />
            </svg>
            <svg
              className="control-button next"
              onClick={[Function]}
              viewBox="0 0 24 24"
            >
              <path
                d="M16,18H18V6H16M6,18L14.5,12L6,6V18Z"
                fill="currentColor"
              />
            </svg>
          </div>
          <div
            className="player-volume"
          >
            <svg
              className="volume-icon"
              fill="currentColor"
              onClick={[Function]}
              viewBox="0 0 20 20"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                clipRule="evenodd"
                d="M9.383 3.076A1 1 0 0110 4v12a1 1 0 01-1.707.707L4.586 13H2a1 1 0 01-1-1V8a1 1 0 011-1h2.586l3.707-3.707a1 1 0 011.09-.217zM14.657 2.929a1 1 0 011.414 0A9.972 9.972 0 0119 10a9.972 9.972 0 01-2.929 7.071 1 1 0 01-1.414-1.414A7.971 7.971 0 0017 10c0-2.21-.894-4.208-2.343-5.657a1 1 0 010-1.414zm-2.829 2.828a1 1 0 011.415 0A5.983 5.983 0 0115 10a5.984 5.984 0 01-1.757 4.243 1 1 0 01-1.415-1.415A3.984 3.984 0 0013 10a3.983 3.983 0 00-1.172-2.828 1 1 0 010-1.415z"
                fillRule="evenodd"
              />
            </svg>
            <div
              className="slider"
            >
              <div
                className="slider-front"
                style={
                  Object {
                    "transform": "translateX(0%)",
                  }
                }
              />
              <input
                className="slider-back"
                onChange={[Function]}
                type="range"
              />
            </div>
          </div>
          <div
            className="player-timeline"
          >
            <div
              className="player-time"
              style={
                Object {
                  "textAlign": "right",
                }
              }
            >
              00
               
              :
               
              00
            </div>
            <div
              className="slider"
            >
              <div
                className="slider-front"
                style={
                  Object {
                    "transform": "translateX(NaN%)",
                  }
                }
              />
              <input
                className="slider-back"
                onChange={[Function]}
                type="range"
              />
            </div>
            <div
              className="player-time"
              style={
                Object {
                  "textAlign": "left",
                }
              }
            >
              00
               
              :
               
              00
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
